File: 2169.Count-Operations-to-Obtain-Zero/solution.py
class Solution(object):
    def countOperations(self, num1, num2):
        """
        :type num1: int
        :type num2: int
        :rtype: int
        """
        if not (num1 and num2):
            return 0
        if num1 >= num2:
            num1 -= num2
            return 1 +  self.countOperations(num1, num2)
        else:
            num2 -= num1
            return 1 + self.countOperations(num1, num2)

File: 2169.Count-Operations-to-Obtain-Zero/solution.ts
function countOperations(num1: number, num2: number): number {
    let solution = 0;

    while (num1 && num2) {
        solution += Math.floor(num1 / num2);
        num1 = num1 % num2;
        [num1, num2] = [num2, num1]
    }
    return solution
};
